Lib/pdb.py

```diff
@@ -2441,9 +2441,12 @@ def main():
             traceback.print_exception(e, colorize=_colorize.can_colorize())
             print("Uncaught exception. Entering post mortem debugging")
             print("Running 'cont' or 'step' will restart the program")
-            pdb.interaction(None, e)
-            print(f"Post mortem debugger finished. The {target} will "
-                  "be restarted")
+            try:
+                pdb.interaction(None, e)
+            except Restart:
+                print("Restarting", target, "with arguments:")
```

Comment:
You could define the restart function to call do_run and then print this.

Reply:
`do_restart` is an alias for `do_run` now. They both raise `Restart`. Do you suggest that I put the print statement inside the command?

Reply:
I'm suggesting that do_restart prints and then calls do_run.

Reply:
So do not print this for `run` command? Because currently, `run` command also prints this line. `restart` and `run` are exactly the same command. And we should differentiate them? `run` also restarts the program.

Reply:
In that case why not just put the print in do_run?

Reply:
There are two concerns:
1. `restart` (and `run`) only works when pdb is brought up from command line (`python -m pdb`). It won't work if the user put `breakpoint()` in the source code. In that case, `Restart` exception is raised and the program exists with an exception. Do we want to print this line in that case?
2. We won't be able to get rid of the except block anyway. There's control flows required in the block. So we can put the prints in a unified place but the except blocks will still be left there.